src/Tools/dotnet-dump/Dumper.cs

```diff
@@ -29,11 +29,32 @@ public Dumper()
         {
         }
 
-        public async Task<int> Collect(IConsole console, int processId, string output, bool diag, DumpTypeOption type)
+        public async Task<int> Collect(IConsole console, int processId, string transportPath, string output, bool diag, DumpTypeOption type)
         {
-            if (processId == 0) {
-                console.Error.WriteLine("ProcessId is required.");
-                return 1;
+            if (string.IsNullOrEmpty(transportPath))
+            {
+                if (processId == 0) {
+                    console.Error.WriteLine("ProcessId is required.");
+                    return 1;
+                }
+            }
+            else
+            {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    Console.Error.WriteLine("Cannot create dumps via trasnportPath on Windows");
+                    return 1;
+                }
+                else if (!File.Exists(transportPath) && !File.Exists(@"\\.pipe\" + transportPath))
```

Comment:
I wouldn't be surprised if someone attempts to pass "\\\\.pipe\something" as the pipe path argument. I would either detect the prefix and help the user out by handling it automatically or include the final formatted path in the error message to clue the user in that the tool is going to prepend \\\\.pipe\ to whatever argument they provide.

Reply:
The client library normalizes the name later on regardless of the prefix being there or not, so it should work with or without `\\.\pipe\` here.  I'll update the error message though.